### ISSUE TITLE
Training with different LR

### ISSUE THREAD

**Issue by Daniel17bn** [Daniel17bn/BSP-RL-Exploration#1]
<img width="3169" height="1970" alt="Image" src="https://github.com/user-attachments/assets/c02f9db2-2de6-43fe-9fa5-4d51663b2f4a" /> 

_total_episodes=50_000,
gamma=0.99,
epsilon_start=1.0,
epsilon_decay=0.99995,
epsilon_min=0.05,
is_rainy=False,
fickle_passenger=False,
action_size=6,
state_size=500_

**Reward per Episode**

Learning Rate 0.1 (blue) and 0.5 (cyan):

- Both converge rapidly to near-optimal performance (~0 reward plateau)

- Initial learning is extremely fast - reaching good performance within ~5,000 episodes

- Almost identical final performance, suggesting both are sufficient for this problem
- The curves are nearly overlapping, indicating robustness across these higher learning rates

Learning Rate 0.01 (orange):

- Significantly slower convergence - takes ~25,000 episodes to reach the same performance
- Starts from much worse initial performance (~-800 reward)
- Eventually reaches similar final performance, but requires 5x more episodes
- Key insight: LR 0.01 is too conservative for this environment - the agent updates its Q-values too slowly

**Episode Lengths**

Learning Rate 0.01:

- Takes much longer to reduce episode length
- Shows the agent is exploring inefficiently for longer periods
- Gradual descent indicates slow policy improvement

**Q-value Changes**

Learning Rate 0.5 (cyan):

- Drops extremely rapidly to near-zero (~10⁻¹⁴) within ~10,000 episodes
- Flatlines early, indicating Q-table has essentially converged
- Risk: Such rapid convergence might indicate premature convergence or numerical instability
- The occasional spikes to 10⁻¹⁷ are likely floating-point precision limits

Learning Rate 0.1 (blue):

- More gradual descent to ~10⁻³ by episode 20,000
- Stabilizes with small ongoing updates (~10⁻⁵ to 10⁻⁸)
- Healthiest convergence pattern - smooth, stable, and thorough
- Ongoing micro-adjustments suggest fine-tuning continues

Learning Rate 0.01 (orange):

- Slowest descent - still at ~10⁻² by episode 50,000
- Never reaches the stability of higher learning rates
- Q-values are still changing significantly at the end of training
- Underfitting: The agent hasn't fully converged even after 50k episodes
